server/bleep/src/webserver/answer.rs

```diff
@@ -71,14 +78,42 @@ pub struct AnswerResponse {
 const SNIPPET_COUNT: usize = 13;
 
 pub async fn handle(
+    params: Query<Params>,
+    Extension(app): Extension<Application>,
+) -> Result<impl IntoResponse, impl IntoResponse> {
+    // create a new analytics event for this query
+    let mut event = QueryEvent::default();
+
+    // populate analytics event
+    let response = _handle(params, Extension(app.clone()), &mut event).await;
```

Comment:
Can you not just straight up send the `app` instance through? Is there any reason the callee needs to take `Extension<Application>`?

Reply:
ah yes, no reason really, we may as well pass `app` here.